games/space-invaders/Space Invaders [David Winter].json

```diff
@@ -0,0 +1,108 @@
+{
+  "title": "Space Invaders",
+  "authors": [
+    {
+      "id": "david-winter",
+      "name": "David Winter"
+    }
+  ],
+  "release": "1978",
+  "category": "GAME",
+  "description": "The well known game. Destroy the invaders with your ship.",
+  "images": [
+    "./Space Invaders [David Winter] - main screen.png",
+    "./Space Invaders [David Winter] - game play.png"
+  ],
+  "interpreter": {
+    "id": "chip8-david-winter",
+    "name": "CHIP8 by David Winter"
+  },
+  "binaries": {
+    "names": [
+      { "value": "Space Invaders [David Winter] - 0.9.1" },
+      { "value": "Space Invaders [David Winter]" },
+
+      { "value": "Space Invaders [David Winter] - 0.9.0" },
+      { "value": "Space Invaders [David Winter] (alt)" },
+      { "value": "INVADERS" }
+    ],
+    "extensions": [
+      "c8",
+      "ch8",
+      ""
+    ],
+    "hashes": [
+      {
+        "type": "SHA256",
+        "value": "00bb7001de52b562d98357d33375406c6d0308466d7d95c2580537d2451a75cd"
+      },
+      {
+        "type": "SHA256",
+        "value": "2d0e1fa53216b297e74041d4fb766f42327a42893e83bb4ec931a9dff5c2dd10"
+      }
+    ]
+  },
+  "sources": {
+    "names": [
+      {
+        "type": "DISASSEMBLED",
+        "value": "Space Invaders [David Winter] - 0.9.0"
+      },
+      {
+        "type": "DISASSEMBLED",
+        "value": "Space Invaders [David Winter] - 0.9.1"
+      }
+    ],
+    "extensions": [
+      "8o",
+      "asm",
+      "txt",
+      ""
+    ]
+  },
+  "platform": {
+    "name": "chip8",
+    "startAddress": "0x200",
+    "resolution": {
+      "width": 64,
+      "height": 32
+    }
+  },
+  "keys": [
+    {
+      "label": "Start",
+      "action": "START",
+      "stroke": "ENTER",
+      "value": "0x5"
+    },
+    {
+      "label": "Fly left",
+      "action": "LEFT",
+      "stroke": "LEFT",
+      "value": "0x4"
+    },
+    {
+      "label": "Fly right",
+      "action": "RIGHT",
+      "stroke": "RIGHT",
+      "value": "0x6"
+    },
+    {
+      "label": "Shoot",
+      "action": "X",
+      "stroke": "SPACE",
+      "value": "0x5"
+    }
+  ],
+  "options": {
+    "opsPerSec": 1500,
+
+    "shiftOp": "MODERN",
+    "loadStoreOp": "UNUSED",
```

Comment:
I'm not really keen to name behavioral differences between specs as quirks. COSMAC VIP interpreter was following "v1.0" of the spec and HP48 followed "v2.0". They are different, one is not better than the other. Also as we discussed (on reddit I think) it's really misleading. Which spec is the proper one and which one is quirky?

Another thing is, some games don't care how shift works because they don't use it. So there is no point in specifying if the quirk should be enabled or disabled (hence true-false is not enough here).

I'm proposing to use following template:
`"nameOfTheOp": "[MODERN|LEGACY|UNUSED]"`

However the values could be different (based on interpreter, or some spec versioning scheme that we create :) ). 

IMHO This part of the metadata file is the hardest to define.

Reply:
> I'm not really keen to name behavioral differences between specs as quirks.

"Quirks" is a term that's used a lot by now, so I think it makes sense to continue using it, but I'm not married to it.

> Another thing is, some games don't care how shift works because they don't use it. So there is no point in specifying if the quirk should be enabled or disabled (hence true-false is not enough here).

My thinking here was that if the game contains an entry for a specific quirk, that means that it cares about the behavior of that quirk. So the values are really true, false and NA/nil/not present. Not sure if that holds up, but that was my thinking.

> COSMAC VIP interpreter was following "v1.0" of the spec and HP48 followed "v2.0".

A problem with thinking in this way (or modern/legacy) is that COSMAC VIP has one behavior for FX55/FX65, CHIP48 (the first CHIP-8 interpreter for HP48) has another, and SCHIP (also for HP48) has a third... We could use something like `"loadStore": "[incrementOncePerAccess | incrementOncePerAccessPlusOne | noIncrement ]"` but it quickly gets messy.

Definitely the hardest part of the schema to define!

Reply:
> "Quirks" is a term that's used a lot by now, so I think it makes sense to continue using it

You are right. I saw this in other contexts as well (GB, NES)

> So the values are really true, false and NA/nil/not present. 

So a `Boolean` which can be `true`, `false`, `null`.

> A problem with thinking in this way (or modern/legacy) is that COSMAC VIP has one behavior for FX55/FX65, CHIP48 (the first CHIP-8 interpreter for HP48) has another, and SCHIP (also for HP48) has a third...

Damn it :/

Reply:
Switched back to quirks. I guess every quirk could have separate set of possible descriptive values (to avoid confusion when interpreting the metadata on the client side)